Tests/SwiftProtobufTests/Test_StreamDecodingIterator.swift

```diff
@@ -0,0 +1,232 @@
+// Tests/SwiftProtobufTests/Test_StreamDecodingIterator.swift - Delimited message tests
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+import Foundation
+import XCTest
+import SwiftProtobuf
+
+class Test_StreamDecodingIterator: XCTestCase, StreamErrorDelegate {
+  
+  var error: Error?
+  var errorExpectation: XCTestExpectation?
+  
+  func onError(error: Error) {
+    self.error = error
+    self.errorExpectation?.fulfill()
+    self.errorExpectation = nil
+  }
+  
+  //Decode zero length messages
+  func testZeroLengthMessages() {
+    var messages = [ProtobufUnittest_TestAllTypes]()
+    for _ in 1...10 {
+      messages.append(ProtobufUnittest_TestAllTypes())
+    }
+    
+    let data = serializeToMemory(messages: messages)
+    XCTAssert(data.count == 10) //Should be 10 zero varints
+    
+    let inputStream = InputStream(data: data)
+    inputStream.open()
+    let iterator = ProtobufUnittest_TestAllTypes.streamDecodingIterator(inputStream: inputStream)
+    var count = 0
+    while let message = iterator.next() {
+      XCTAssertEqual(message, ProtobufUnittest_TestAllTypes())
+      count += 1
+    }
+    XCTAssert(count == 10)
+    inputStream.close()
+  }
+  
+  //Empty stream with no bytes
+  func testEmptyStream() {
+    let inputStream = InputStream(data: Data(count: 0))
+    inputStream.open()
+    let iterator = ProtobufUnittest_TestAllTypes.streamDecodingIterator(inputStream: inputStream)
+    while let _ = iterator.next() {
+      XCTFail("Stream was empty")
```

Comment:
Probably want to clarify this to be something like - Shouldn't have returned a value for an empty stream.

Reply:
Done